packages/drivers/routerlicious-driver/src/test/cache.spec.ts

```diff
@@ -0,0 +1,51 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { strict as assert } from "assert";
+import { SinonFakeTimers, useFakeTimers } from "sinon";
+import { ICache, InMemoryCache } from "../cache";
+
+describe("InMemoryCache", () => {
+    let clock: SinonFakeTimers;
+    let cache: ICache<number>;
+
+    before(() => {
+        clock = useFakeTimers();
+    });
+
+    after(() => {
+        clock.restore();
+    });
+
+    it("No expiry", async () => {
+        cache = new InMemoryCache();
+        await cache.put("one", 1);
+
+        clock.tick(Number.MAX_SAFE_INTEGER);
+
+        assert.equal(await cache.get("one"), 1, "Entry shouldn't expire");
+    });
+
+    it("With expiry", async () => {
+        const expiryMs = 10;
+        cache = new InMemoryCache<number>(expiryMs);
+
+        await cache.put("one", 1);
+        await cache.put("two", 2);
+
+        clock.tick(5);
+        assert.equal(await cache.get("one"), 1, "'one' shouldn't be expired after 5ms");
+        assert.equal(await cache.get("two"), 2, "'two' shouldn't be expired after 5ms");
+
+        await cache.get("one"); // Should NOT reset the expiry, only put does
+        await cache.put("two", 2.1);
+        await cache.put("three", 3);
+
+        clock.tick(5);
+        assert.equal(await cache.get("one"), 1, "'one' should be expired after 10ms");
```

Comment:
Shouldn't this be undefined since the cache expired?

Reply:
I forgot to reset the clock after each test so current time was NaN and it broke assumptions.